src/osx/core/bitmap.cpp

```diff
@@ -364,57 +342,91 @@ bool wxBitmapRefData::Create(CGContextRef context)
 
 bool wxBitmapRefData::Create( int w , int h , int d, double logicalscale )
 {
+#if wxOSX_BITMAP_NATIVE_ACCESS
+    int m_width, m_height, m_depth, m_bytesPerRow;
+#endif
     m_width = wxMax(1, w);
     m_height = wxMax(1, h);
     m_depth = d ;
     m_scaleFactor = logicalscale;
     m_hBitmap = NULL ;
 
     m_bytesPerRow = GetBestBytesPerRow( m_width * 4 ) ;
+#if !wxOSX_BITMAP_NATIVE_ACCESS
     size_t size = m_bytesPerRow * m_height ;
     void* data = m_memBuf.GetWriteBuf( size ) ;
     if ( data != NULL )
     {
         memset( data , 0 , size ) ;
         m_memBuf.UngetWriteBuf( size ) ;
-
+#else
+        void* data = NULL;
+#endif
         m_hBitmap = CGBitmapContextCreate((char*) data, m_width, m_height, 8, m_bytesPerRow, wxMacGetGenericRGBColorSpace(), kCGImageAlphaNoneSkipFirst );
         wxASSERT_MSG( m_hBitmap , wxT("Unable to create CGBitmapContext context") ) ;
         CGContextTranslateCTM( m_hBitmap, 0,  m_height );
-        CGContextScaleCTM( m_hBitmap, 1*m_scaleFactor, -1*m_scaleFactor );
+        CGContextScaleCTM( m_hBitmap, 1*GetScaleFactor(), -1*GetScaleFactor() );
+#if !wxOSX_BITMAP_NATIVE_ACCESS
     } /* data != NULL */
+#endif
     m_ok = ( m_hBitmap != NULL ) ;
 
     return m_ok ;
 }
 
 void wxBitmapRefData::UseAlpha( bool use )
 {
-    if ( m_hasAlpha == use )
+    if ( HasAlpha() == use )
         return ;
 
+#if !wxOSX_BITMAP_NATIVE_ACCESS
     m_hasAlpha = use ;
+    void *data = m_memBuf.GetData();
+#else
+    void *data = NULL;
+#endif
+    CGContextRef hBitmap = CGBitmapContextCreate(data, GetWidth(), GetHeight(), 8, GetBytesPerRow(), wxMacGetGenericRGBColorSpace(), use ? kCGImageAlphaPremultipliedFirst : kCGImageAlphaNoneSkipFirst );
+#if wxOSX_BITMAP_NATIVE_ACCESS
+    memcpy(CGBitmapContextGetData(hBitmap),CGBitmapContextGetData(m_hBitmap),GetBytesPerRow()*GetHeight());
+#endif
+    wxASSERT_MSG( hBitmap , wxT("Unable to create CGBitmapContext context") ) ;
+    CGContextTranslateCTM( hBitmap, 0,  GetHeight() );
+    CGContextScaleCTM( hBitmap, 1*GetScaleFactor(), -1*GetScaleFactor() );
```

Comment:
Minor, but why `1*`? I find this more confusing than helpful...

Reply:
True, changed